### PR TITLE
prepare message timeout functionality

### LINKED ISSUE #1079: sendTo will not reject / fail if instance does not exist
Hi there,

I'm bulk requesting with sendTo (https://github.com/ioBroker/ioBroker.js-controller/blob/master/lib/adapter.js#L5457) in the following way (using `allSettled`):
```
new Promise((resolve, reject) => socket.sendTo('getEnabledDPs', null, 'history.0', (err, state) => err ? reject(err) : resolve(state))),
new Promise((resolve, reject) => socket.sendTo('getEnabledDPs', null, 'influxdb.0', (err, state) => err ? reject(err) : resolve(state))),
new Promise((resolve, reject) => socket.sendTo('getEnabledDPs', null, 'sql.0', (err, state) => err ? reject(err) : resolve(state)))
```

My challenge is that when an adapter instance does not exist, I will not receive an answer. Currently I'm addressing this by setting a timeout, but depending on the time of the request / loading time, etc. this may lead to wrong behavior.

Could you implement a rejection in case the receiving instance does not exist? Thanks!

### PR DESCRIPTION
**Pull request by foxriver76**
- closes #1079

### CODE REVIEW

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 1]
```diff
@@ -586,8 +589,8 @@ export class AdapterClass extends EventEmitter {
     private eventLoopLags: number[] = [];
     private overwriteLogLevel: boolean = false;
     adapterReady: boolean = false;
-    private callbacks?: Record<string, { cb: ioBroker.MessageCallback; time?: number }>;
-
+    /** Callbacks from sendTo */
+    private readonly callbacks = new Map<number, MessageCallbackObject>();
```
Comment (Apollon77) [inline packages/adapter/src/lib/adapter/adapter.ts:593]: we could name it a bit more specific then "callbacks" ... could be anything :-)

**Review by Apollon77 — COMMENTED** [review]
I think we should also cleanup ALL callback timer on stop, or?

**Comment by foxriver76** [comment]
> I think we should also cleanup ALL callback timer on stop, or?

Yes, it is already in https://github.com/ioBroker/ioBroker.js-controller/pull/2252/files#diff-fdf15c6340b088e24a291222e4d17edc0b96c2dde90b77d0f2a499dd24fcd425R2125-R2128

Or do you mean something additional?

**Review by Apollon77 — APPROVED** [review]
no,,seems I mised that place in review

**Pull request merged**